adobe/algorithm/selection_algorithms.hpp

```diff
@@ -778,36 +778,127 @@ selection_stable_partition_about(const Selection& selection, ForwardRange& range
 
 /****************************************************************************************************/
 /*!
-    \ingroup selection_algorithms
-
-    Takes a set of indices and converts them to a boundary-based Selection
-*/
+ \ingroup selection_algorithms
+ 
+ OLD VERSION (before September 2016) Takes a set of indices and converts them to a boundary-based Selection
+ */
 template <typename Selection, typename ForwardRange>
-Selection index_set_to_selection(const ForwardRange& index_set) {
+Selection index_set_to_selection_old(const ForwardRange& index_set) {
     Selection result;
-
+    
     // REVISIT (fbrereto) : This would go much faster using divide-and-conquer
     //                      and eventually balanced reduction.
-
+    
     typedef typename boost::range_const_iterator<ForwardRange>::type range_const_iterator;
-
+    
     range_const_iterator iter(boost::begin(index_set));
     range_const_iterator last(boost::end(index_set));
-
+    
     for (; iter != last; ++iter) {
         Selection tmp;
-
+      
         tmp.push_back(*iter);
         tmp.push_back(*iter + 1);
-
+      
         result = selection_union(result, tmp);
     }
-
+    
     return result;
 }
 
 /****************************************************************************************************/
 /*!
+ \ingroup selection_algorithms
+ 
+ Finds end of the non-interrupted sequence in a strictly increasing set of indices, starting with number n
+ 
+ \pre [f, l) is a sorted (strictly increasing) set of indices
+ 
+ \return
+ A pair of values. The first is a iterator for the next element of a non-interupted sequence
+ The second is the last number of a non-interrupted sequence .
+ 
+ E.g. in an array of [0, 1, 2, 3, 7, 9, 10, 12]
+  : when called with value of ((iter 1), (iter 12), 1) : returns {(iter 7), 3};
+  : when called with value of ((iter 9), (iter 12), 8) : returns {(iter 9), 8};
+ 
+ */
+template <typename I, typename N> // I models ForwardIterator
+std::pair<I, N> find_sequence_end(I f, I l, N n) {
+    while (f != l && n == *f) {
+        ++n;
+        ++f;
+    }
+    return { f, n };
+}
+ 
+  
+/****************************************************************************************************/
+/*!
+  \ingroup selection_algorithms
+
+  Takes an subset of strictly increasing set of indices and converts them to a boundary-based Selection
+
+  \pre [f, l) is a sorted (strictly increasing) set of indices
+
+  \return
+  A Selection.
+*/
+
+template <typename Selection, typename I> // I models ForwardIterator
+Selection strictly_increasing_index_set_to_selection(I f, I l) {
```

Comment:
Since this is just the iterator form of the below function it should have the same name. I also wouldn't make Selection a template parameter, either use selection_t or use an output iterator - to call the output iterator form with a selection_t it would look like:

selection_t x;
index_set_to_selection(index_set, std::back_inserter(x));